ansible-ipi-install/roles/bootstrap/tasks/10_load_inv.yml

```diff
@@ -189,110 +200,103 @@
       set_fact:
-        supermicro_workers: "{{ supermicro_workers | default([]) + [ item ] }}"
+        supermicro_workers: "{{ supermicro_workers + [ item ] }}"
       with_items:
         - "{{ worker_fqdns }}"
       when: lab_name == 'alias' and item.split('.')[0].split('-')[2] in lab_vars['machine_types']['supermicro']
 
+    # there are always 3 masters so master_fqdns is always non-empty
+    # and loop will always be run so master_types is always defined
+ 
     - name: Set master node types
       set_fact:
-        master_types: "{{ master_types | default([]) + [ (lab_name == 'scale') | ternary(item.split('.')[0].split('-')[3], item.split('.')[0].split('-')[2]) ] }}"
+        master_types: "{{ master_types + [ (lab_name == 'scale') | ternary(item.split('.')[0].split('-')[3], item.split('.')[0].split('-')[2]) ] }}"
       with_items:
         - "{{ master_fqdns }}"
 
     - name: Set master node dells (scale)
       set_fact:
-        dell_masters: "{{ dell_masters | default([]) + [ item ] }}"
+        dell_masters: "{{ dell_masters + [ item ] }}"
       with_items:
         - "{{ master_fqdns }}"
       when: lab_name == 'scale' and item.split('.')[0].split('-')[3] in lab_vars['machine_types']['dell']
 
     - name: Set master node dells (alias)
       set_fact:
-        dell_masters: "{{ dell_masters | default([]) + [ item ] }}"
+        dell_masters: "{{ dell_masters + [ item ] }}"
       with_items:
         - "{{ master_fqdns }}"
       when: lab_name == 'alias' and item.split('.')[0].split('-')[2] in lab_vars['machine_types']['dell']
 
     - name: Set master node supermicros (scale)
       set_fact:
-        supermicro_masters: "{{ supermicro_masters | default([]) + [ item ] }}"
+        supermicro_masters: "{{ supermicro_masters + [ item ] }}"
       with_items:
         - "{{ master_fqdns }}"
       when: lab_name == 'scale' and item.split('.')[0].split('-')[3] in lab_vars['machine_types']['supermicro']
 
     - name: Set master node supermicros (alias)
       set_fact:
-        supermicro_masters: "{{ supermicro_masters | default([]) + [ item ] }}"
+        supermicro_masters: "{{ supermicro_masters + [ item ] }}"
       with_items:
         - "{{ master_fqdns }}"
       when: lab_name == 'alias' and item.split('.')[0].split('-')[2] in lab_vars['machine_types']['supermicro']
 
     - name: Set Dell nodes
       set_fact:
-        dell_nodes: "{{ dell_masters | default([]) + dell_workers | default([])  }}"
+        dell_nodes: "{{ dell_masters + dell_workers }}"
 
     - name: Set Supermicro nodes
       set_fact:
-        supermicro_nodes: "{{ supermicro_masters | default([]) + supermicro_workers | default([])  }}"
+        supermicro_nodes: "{{ supermicro_masters + supermicro_workers }}"
   vars:
     lab_vars: "{{ (lab_name == 'scale') | ternary(scale, alias) }}"
   when: lab_name in ['scale', 'alias']
 
-- name: Fail when masters are non homogenuous or don't match provisioner
```

Comment:
@bengland2 as discussed this is a much needed condition to avoid installer failure, master provisioning interface is expected to be same across. https://github.com/redhat-performance/JetSki/blob/master/ansible-ipi-install/roles/installer/templates/install-config.j2#L41
In rare cases like yours, you need to manually comment this out in your local copy. 
@smalleni any other thoughts and comments ?

Reply:
I agree, pushed a change reinserting the homogeneity test.